src/combinations.rs

```diff
@@ -52,7 +52,16 @@ pub trait PoolIndex<T>: BorrowMut<[usize]> {
         self.borrow().len()
     }
 }
+impl<T> PoolIndex<T> for Box<[usize]> {
+    type Item = Vec<T>;
```

Comment:
We can think about returning `Box<[T]>` later, but since we'd probably construct it via a `Vec`, `Vec` is fine.